src/main/java/io/vertx/codegen/TypeValidator.java

```diff
@@ -111,6 +116,16 @@ private static boolean isValidNonCallableType(Element elem, TypeInfo type, boole
     return false;
   }
 
+  private static boolean isValidKotlinCompanionObjet(Element elem, TypeInfo type) {
+    String qualifiedName = type.getName(); // e.g. io.vertx.test.codegen.testapi.kotlin.InterfaceWithCompanionObject.Companion
+    String simpleName = type.getSimpleName(); // e.g. Companion
+    String enclosingElementName = elem.getEnclosingElement().getSimpleName().toString(); // e.g. InterfaceWithCompanionObject
+
+    // Ensure the object is declared in the enclosing element.
+    // (This only checks the last 2 segments; ideally we'd check against the enclosing element's qualified name.)
+    return qualifiedName.endsWith(enclosingElementName + "." + simpleName);
```

Comment:
I previously used `modelElt.getQualifiedName().toString()` to get the qualified name of the enclosing element.

Let me know if there's an equivalent for the new model system.